src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/PostgreSqlEngine.java

```diff
@@ -126,6 +132,25 @@ protected int entityToPreparedStatement(final DbEntity entity, final PreparedSta
     }
 
     @Override
+    public synchronized void setParameter(final String name, final int index, Object param, DbColumnType paramType) throws DatabaseEngineException, ConnectionResetException {
+        if (paramType == DbColumnType.JSON) {
+            param = getJSONValue((String)param);
+        }
+        super.setParameter(name, index, param);
+    }
+
+    private Object getJSONValue(String val) throws DatabaseEngineException {
+        try {
+            PGobject dataObject = new PGobject();
+            dataObject.setType("jsonb");
+            dataObject.setValue(val);
+            return dataObject;
+        } catch (SQLException ex) {
+            throw new DatabaseEngineException("Error while mapping variables to database", ex);
```

Comment:
You have the value, could be worthwhile including it in the exception for more context.